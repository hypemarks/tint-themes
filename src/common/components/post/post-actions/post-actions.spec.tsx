import { render } from '@testing-library/react';
import React from 'react';
import PostActions from './post-actions';
import { buildPost } from '../../../../../test/generate';

describe('Test <PostActions /> Component', () => {
  it('should render snapshot', () => {
    const post = buildPost();
    const wrapper = render(<PostActions externalId={post.attributes.external_id} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
