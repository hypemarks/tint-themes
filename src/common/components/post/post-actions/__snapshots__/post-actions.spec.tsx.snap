// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PostActions /> Component should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="post-actions"
    data-testid="post-actions"
  >
    <div
      class="post-actions__action"
    >
      <a
        aria-label="Reply to this tweet. Opens in new window."
        data-testid="link"
        href="https://twitter.com/intent/tweet?in_reply_to=123"
        rel="noopener noreferrer"
        tabindex="0"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-retweet fa-w-20 fa-sm "
          data-icon="retweet"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M629.657 343.598L528.971 444.284c-9.373 9.372-24.568 9.372-33.941 0L394.343 343.598c-9.373-9.373-9.373-24.569 0-33.941l10.823-10.823c9.562-9.562 25.133-9.34 34.419.492L480 342.118V160H292.451a24.005 24.005 0 0 1-16.971-7.029l-16-16C244.361 121.851 255.069 96 276.451 96H520c13.255 0 24 10.745 24 24v222.118l40.416-42.792c9.285-9.831 24.856-10.054 34.419-.492l10.823 10.823c9.372 9.372 9.372 24.569-.001 33.941zm-265.138 15.431A23.999 23.999 0 0 0 347.548 352H160V169.881l40.416 42.792c9.286 9.831 24.856 10.054 34.419.491l10.822-10.822c9.373-9.373 9.373-24.569 0-33.941L144.971 67.716c-9.373-9.373-24.569-9.373-33.941 0L10.343 168.402c-9.373 9.373-9.373 24.569 0 33.941l10.822 10.822c9.562 9.562 25.133 9.34 34.419-.491L96 169.881V392c0 13.255 10.745 24 24 24h243.549c21.382 0 32.09-25.851 16.971-40.971l-16.001-16z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
    <div
      class="post-actions__action"
    >
      <a
        aria-label="Re-tweet this tweet. Opens in new window."
        data-testid="link"
        href="https://twitter.com/intent/retweet?tweet_id=123"
        rel="noopener noreferrer"
        tabindex="0"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-reply fa-w-16 fa-sm "
          data-icon="reply"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.309 189.836L184.313 37.851C199.719 24.546 224 35.347 224 56.015v80.053c160.629 1.839 288 34.032 288 186.258 0 61.441-39.581 122.309-83.333 154.132-13.653 9.931-33.111-2.533-28.077-18.631 45.344-145.012-21.507-183.51-176.59-185.742V360c0 20.7-24.3 31.453-39.687 18.164l-176.004-152c-11.071-9.562-11.086-26.753 0-36.328z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
    <div
      class="post-actions__action"
    >
      <a
        aria-label="Like this tweet. Opens in new window."
        data-testid="link"
        href="https://twitter.com/intent/favorite?tweet_id=123"
        rel="noopener noreferrer"
        tabindex="0"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-heart fa-w-16 fa-sm "
          data-icon="heart"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
